src/GovUk.Education.ExploreEducationStatistics.Admin/Services/ReleaseDataFileService.cs

```diff
@@ -282,6 +282,10 @@ await _releaseVersionRepository
                                 replacingDataFile: replacingFile,
                                 order: releaseDataFileOrder);
 
+                            var dataReleaseFile = await _releaseFileRepository.Find(
+                                releaseVersionId: releaseVersionId,
+                                fileId: dataFile.Id);
```

Comment:
This returns an optional `ReleaseFile`, and I don't think the `ReleaseFile.File` is guaranteed to be included in the result of it, but is depended on by the `DataFileViewModel` method.

Just thinking it might be better to use the `ContentDbContext` variable directly here and remove these couple of potential issues with `Find`?

Reply:
I'm undecided on this one. I was also thinking it was weird to use `_releaseFileRepository.Find` as it returns a nullable `ReleaseFile`. However, I thought it was probably OK because in this context both the `ReleaseFile` and `File` **should** exist. So I was happy to let it blow up if it was `null`, because it shouldn't ever happen. So In the end I decided to still use it, as it's nice to re-use the existing method.

But I also get why you'd want to use `ContextDbContext` directly to be more explicit, and I could use `.Single` rather than `.SingleOrDefault`.

Reply:
Yes I think it might be better to switchover or at least refactor it slightly so that you can specify what's included?

The `BuildDataFileViewModel` has a lot of references to `dataReleaseFile.File`. If that's working, I think it's only going to be because the `ReleaseFileRepository.Find` method has the line `.Include(releaseFile => releaseFile.File.CreatedBy)` which I'm guessing means the whole of `releaseFile.File` is being hydrated.

Reply:
Yeah it's currently working because of that line, you're right.

I've decided to make it more explicit and write the query inline! :)